src/simd/distances_avx.cc

```diff
@@ -38,80 +40,34 @@ masked_read(int d, const float* x) {
     // cannot use AVX2 _mm_mask_set1_epi32
 }
 
+// trust the compiler to unroll this properly
+FAISS_PRAGMA_IMPRECISE_FUNCTION_BEGIN
```

Comment:
1. auto-vectorizers exists in almost every compiler. The quality depends on the compiler version & family, but for a simple code like this one every compiler should work
2. depending on the flags that are used for compiling a particular file (`-mavx`, `-mavx2`, `-mavx512f`, `-mfma`)

Reply:
Hi Alex,
Currently when knowhere init, it will select SIMD API sets by config or AUTO, it can guarantee that Milvus can run on all kinds of intel CPUs.
If we use auto-vectorizer, seems we need not src/simd/hook.cc any more, is it correct ?

Reply:
Please let me know which compiler versions are going to be used.